tools/ci_connector_ops/ci_connector_ops/pipelines/commands/metadata_service.py

```diff
@@ -0,0 +1,73 @@
+import sys
+import click
+import anyio
+import dagger
+import logging
+
+from ci_connector_ops.pipelines.bases import Step, StepStatus, TestReport
+from ci_connector_ops.pipelines.actions import environments
+from ci_connector_ops.pipelines.contexts import PipelineContext
+from ci_connector_ops.pipelines.utils import (
+    DAGGER_CONFIG,
+    with_exit_code,
+)
+from rich.logging import RichHandler
+
+
+logging.basicConfig(level=logging.INFO, format="%(name)s: %(message)s", datefmt="[%X]", handlers=[RichHandler(rich_tracebacks=True)])
+logger = logging.getLogger(__name__)
+
+METADATA_LIB_MODULE_PATH = "airbyte-ci/connectors/metadata_service/lib"
+
+
+class MetadataLibRunTest(Step):
```

Comment:
@alafanechere putting this up now slightly incomplete (needs some comments and classes moved to new files) just to get your thoughts on this.

Generally do you feel like im using the classes you defined in the correct spirit of things?

Also I had an idea while writing this. What if we made the PipelineContext hold onto a collection of steps and the be responsible for running them?

```python
async def run_metadata_pipeline(cli_context):
    async with dagger.Connection(DAGGER_CONFIG) as dagger_client:
        metadata_pipleline_context = PipelineContext(
            pipeline_name="Metadata Service Lib Unit Test Pipeline",
            is_local=cli_context.obj["is_local"],
            git_branch=cli_context.obj["git_branch"],
            git_revision=cli_context.obj["git_revision"],
            gha_workflow_run_url=cli_context.obj.get("gha_workflow_run_url"),
            pipeline_start_timestamp=cli_context.obj.get("pipeline_start_timestamp"),
            ci_context=cli_context.obj.get("ci_context")
            dagger_client = dagger_client
            steps = [
                MetadataLibRunTest,
            ]
        )

        async with metadata_pipleline_context:
            test_report = await metadata_pipleline_context.run()
            return test_report
            ```
Where the pipeline context becomes concerned with everything that happens around a pipeline run
1. Logging
2. Reporting status to github
3. Ordering steps
4. Reporting outcome

Reply:
> Also I had an idea while writing this. What if we made the PipelineContext hold onto a collection of steps and the be responsible for running them?

I like it but it would mean implementing a PipelineContext for each pipeline type and PipelineContext would need to be abstract (as it was before). I feel like its too big of a lift to make this happen in the current PR as it will change quite a lot of things in your PR for adapting the connector pipeline, and in this [PR](https://github.com/airbytehq/airbyte/pull/24225). 
Would you mind creating an issue on the "Connectors CI in production" epic to track this future improvement?

Reply:
Added the issue here: [Update Dagger pipeline class to manage the running of steps](https://app.zenhub.com/workspaces/connector-operations-6318ede5eda32e0010ef8566/issues/gh/airbytehq/airbyte/24752)